tests/store/_unity_catalog/oss_registry/test_oss_unity_catalog_rest_store.py

```diff
@@ -0,0 +1,80 @@
+import json
+from unittest import mock
+
+import pytest
+
+from mlflow.entities.model_registry import RegisteredModelTag
+from mlflow.protos.unity_catalog_oss_messages_pb2 import (
+    CreateRegisteredModel,
+    RegisteredModelInfo,
+    TagKeyValue,
+)
+from mlflow.store._unity_catalog.oss_registry.oss_rest_store import OssUnityCatalogStore
+from mlflow.utils.proto_json_utils import message_to_json
+
+from tests.helper_functions import mock_http_200
+from tests.store._unity_catalog.conftest import _REGISTRY_HOST_CREDS
+
+
+@pytest.fixture
+def store(mock_databricks_uc_oss_host_creds):
+    with mock.patch("mlflow.utils.databricks_utils.get_databricks_host_creds"):
+        return OssUnityCatalogStore(store_uri="databricks-uc")
+
+
+def _args(endpoint, method, json_body, host_creds, extra_headers):
+    res = {
+        "host_creds": host_creds,
+        "endpoint": f"/api/2.0/unity-catalog/{endpoint}",
+        "method": method,
+        "extra_headers": extra_headers,
+    }
+    if extra_headers is None:
+        del res["extra_headers"]
+    if method == "GET":
+        res["params"] = json.loads(json_body)
+    else:
+        res["json"] = json.loads(json_body)
+    return res
+
+
+def _verify_requests(
+    http_request,
+    endpoint,
+    method,
+    proto_message,
+    host_creds=_REGISTRY_HOST_CREDS,
+    extra_headers=None,
+):
+    json_body = message_to_json(proto_message)
+    call_args = _args(endpoint, method, json_body, host_creds, extra_headers)
+    http_request.assert_any_call(**call_args)
+
+
+@mock_http_200
+def test_create_registered_model(mock_http, store):
```

Comment:
unit test lgtm(looks like it will work and use your oss rest store correctly).

can we hook this code to work against a UC service that has already implemented the API? (i think databricks implementation of UC should do)

Reply:
Made a notebook to test the changes [here](https://e2-dogfood.staging.cloud.databricks.com/?o=6051921418418893#notebook/2215403683779572).